src/nvim/msgpack_rpc/channel.c

```diff
@@ -261,7 +255,7 @@ Object channel_send_call(uint64_t id,
     free_channel(channel);
   }
 
-  if (!pending_requests) {
+  if (!channel->pending_requests) {
     send_delayed_notifications();
```

Comment:
This function sends notifications for **all** channels, but does not check `pending_requests == 0` for every channel that has a delayed notification.